ui/src/SensitiveData.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from "react";
+import "@fontsource/roboto";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from "@material-ui/core";
+
+
+class SensitiveData extends Component {
+  state = {
+    columnToUpdate: "",
+    newValue: "",
+    newValueValidationError: "",
+    showDialog: false,
+    allowableSensitiveDataColumns: []
+  };
+
+  openDialog = () => {
+    this.refreshDataFromBackend()
+    this.setState({
+      showDialog: true,
+    });
+  };
+
+  closeDialog = () => {
+    this.setState({
+      columnToUpdate: "",
+      newValue: "",
+      newValueValidationError: "",
+      showDialog: false,
+    });
+  };
+
+  getSensitiveSampleColumns = async () => {
+    const response = await fetch(`/api/surveys/${this.props.surveyId}`);
+    if (!response.ok) {
+      return;
+    }
+
+    const surveyJson = await response.json();
+    const sensitiveColumns = surveyJson.sampleValidationRules
+      .filter((rule) => rule.sensitive)
+      .map((rule) => rule.columnName);
+
+    return sensitiveColumns;
+  };
+
+  onSensitiveDataColumnChange = (event) => {
+    this.setState({ columnToUpdate: event.target.value });
+  };
+
+  onChangeValue = (event) => {
+    this.setState({
+      newValue: event.target.value,
+    });
+  };
+
+  refreshDataFromBackend = async () => {
+    const sensitiveDataColumns = await this.getSensitiveSampleColumns(this.props.surveyId
+    ).then((sensitiveColumns) => {
+      return sensitiveColumns;
+    });
+
+    this.setState({
+      allowableSensitiveDataColumns: sensitiveDataColumns,
+    });
+  };
+
+  onCreate = async () => {
+    const updateSampleSensitive = {
+      caseId: this.props.caseId,
+      sampleSensitive: { [this.state.columnToUpdate]: this.state.newValue }
+    }
+
+    const response = await fetch(
+      `/api/cases/${this.props.caseId}/action/updateSensitiveField`,
+      {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(updateSampleSensitive),
+      }
+    );
+
+    if (response.ok) {
+      alert('Update msg sent successfully')
```

Comment:
this alert shouldn't have been left in the code. You should use the debugger rather than putting alerts in the code